### PR TITLE
feat(ESSNTL-3754): Add Unleash support, with fix

### PR DESCRIPTION
**Pull request by kruai**
# Overview

**NOTE: The only changes from the original PR are from [this commit](https://github.com/RedHatInsights/insights-host-inventory/pull/1269/commits/c3bf5d6fdf555d9810ba0176e0cb03af7e215c29) onwards.**

This PR is being created to address [ESSNTL-3754](https://issues.redhat.com/browse/ESSNTL-3754).
This re-implements #1259, which was reverted by #1268. In addition to undoing that reversion, it configures a cache directory for Unleash in Clowder environments. It uses the `/tmp/.unleashcache` folder by default; if we decide we want to configure this path later, we can add `UNLEASH_CACHE_DIR` to our clowdapp config.

For non-clowder environments, it leaves `UNLEASH_CACHE_DIRECTORY` unset; this makes it use fcache to decide, which is generally recommended (when there aren't permission issues). I got this config info from the flask-unleash docs [here](https://docs.getunleash.io/Flask-Unleash/#configuration).

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [x] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,6 +14,28 @@ The `python-snappy` package requires the core Snappy library to be installed on
 sudo dnf install snappy
 ```
 
+### pg_config
```
Comment (thearifismail) [inline README.md:17]: Why are these needed now but not before?
Couldn't these be added to the Pipfile?

File: lib/feature_flags.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+FLAG_FALLBACK_VALUES = {FLAG_INVENTORY_GROUPS: True}
+
+
+def init_unleash_app(app):
+    UNLEASH.init_app(app)
+
+
+# Raise an error if the toggle is not found on the configured Unleash server.
+# Without this fallback function, is_enabled just returns False without error.
+def custom_fallback(feature_name: str, context: dict) -> bool:
```
Comment (thearifismail) [inline lib/feature_flags.py:21]: shouldn't this function be called default_fallback?

**Review by thearifismail — APPROVED** [review]
LGTM.  A couple of minor comments though

File: README.md (hunk 1 above)
Comment (kruai) [reply README.md:17]: I explained this in the [original PR that we merged](https://github.com/RedHatInsights/insights-host-inventory/pull/1259#issuecomment-1414225703). But no, it can't be added to the pipfile because it's not python-related

File: lib/feature_flags.py (hunk 2 above)
Comment (kruai) [reply lib/feature_flags.py:21]: `default_fallback` is kinda redundant imo, since we could refer to that value as either a default value or a fallback value. I called it `custom_fallback` because that's the name they used in the official documentation, and because it's a custom function that's providing a fallback value

File: swagger/api.spec.yaml [hunk 3]
```diff
@@ -708,6 +708,17 @@ paths:
         '403':
             description: Forbidden
 
+  '/unleash/inventory_groups_toggle':
+    get:
+      summary: Check whether the Groups feature is enabled
+      description: Returns the value of the hbi.api.inventory-groups toggle, if able to be retrieved.
```
Comment (fstavela) [inline swagger/api.spec.yaml:714]: `tags` is missing in this endpoint, look at https://github.com/RedHatInsights/insights-host-inventory/blob/master/swagger/api.spec.yaml#L644 for example. Here it should be probably set to `unleash`.

**Comment by computercamplove** [comment]
@kruai currently i get `None` for response while testing in ephemeral, since unleash is not working for ephemeral could we get `True` instead?

**Review by thearifismail — APPROVED** [review]
LGTM

**Review by thearifismail — APPROVED** [review]
LGTM
